tonilopezmr/CaptchaPartOne.kt

```diff
@@ -0,0 +1,46 @@
+package day1
+
+import org.junit.Test
+import org.junit.Assert.*
+
+
+
+class CaptchaPartOne {
+
+    fun sequence(sequence: String) : Int {
+        val captcha = (sequence + sequence[0])
+        return captcha.foldIndexed(0) { idx, acc, a ->
+            if (idx < sequence.length && a.toString().toInt() == captcha[idx +1].toString().toInt()) {
+                acc + a.toString().toInt()
+            } else {
+                acc
+            }
+        }
+    }
```

Comment:
At first I though of a Fold/Zip solution too but it involved working with indexes and I'm not a big fan of it.

Just one thing about this method. You could map all strings to int before folding, that way you would avoid calling `x.toString().toInt()` 3 times. Casting is one of the most expensive operations, but in this case it wouldn't impact that much on performance probably.

Reply:
Yes! I know but I don't know in a big input or big operation what is better, if two iterations or casting 🤔

Reply:
In this case the difference between casting or iterating one more time won't make a difference 🤷‍♂️